narwhal/test-utils/src/lib.rs

```diff
@@ -553,14 +565,35 @@ pub fn this_cert_parents_with_slow_nodes(
 
             if should_include {
                 parents.insert(parent.digest());
+                total_stake += authority.stake();
+            } else {
+                not_included.push(parent);
             }
         } else {
             // just add it directly as it is not within the slow nodes or we are the
             // same author.
             parents.insert(parent.digest());
+            total_stake += authority.stake();
         }
     }
 
+    // ensure we'll have enough parents (2f + 1)
+    while total_stake < committee.quorum_threshold() {
+        let parent = not_included.pop().unwrap();
```

Comment:
Is the order of `not_included` randomized?

Reply:
It isn't , but is not like a hard requirement to have it randomized - same as the original ancestors list, we don't randomise within this method before we process.